lib/shared/bucketing-assembly-script/assembly/bucketing/bucketing.ts

```diff
@@ -196,38 +229,77 @@ export function bucketUserForVariation(
     }
 }
 
+class BucketedFeature {
+    feature: Feature
+    variation: Variation
+}
+
 export function _generateBucketedConfig(
     config: ConfigBody,
     user: DVCPopulatedUser,
-    clientCustomData: JSON.Obj
+    clientCustomData: JSON.Obj,
+    overrides: JSON.Obj | null,
 ): BucketedUserConfig {
     const variableMap = new Map<string, SDKVariable>()
     const featureKeyMap = new Map<string, SDKFeature>()
     const featureVariationMap = new Map<string, string>()
     const variableVariationMap = new Map<string, FeatureVariation>()
 
+    const bucketedFeatureVariations: Array<BucketedFeature> = []
+
     for (let i = 0; i < config.features.length; i++) {
         const feature = config.features[i]
         const targetAndHashes = doesUserQualifyForFeature(
             config,
             feature,
             user,
-            clientCustomData
+            clientCustomData,
         )
 
         if (!targetAndHashes) continue
 
         const variation = bucketUserForVariation(feature, targetAndHashes)
 
-        featureKeyMap.set(feature.key, new SDKFeature(
-            feature._id,
-            feature.type,
+        bucketedFeatureVariations.push({ feature, variation })
+    }
+
+    if (overrides != null) {
+        const overrideKeys = overrides.keys
+        for (let i = 0; i < overrideKeys.length; i++) {
+            const featureId = overrideKeys[i]
+            const feature = config.getFeatureForId(featureId)
+            if (feature == null) {
+                continue
+            }
+            const variationId = getStringFromJSON(overrides, featureId)
+
+            const variation = feature.getVariationById(variationId)
+
+            if (variation == null) {
+                continue
+            }
+
+            bucketedFeatureVariations.push({ feature, variation })
+        }
+    }
+
+    for (let i = 0; i < bucketedFeatureVariations.length; i++) {
```

Comment:
Ideally, we could structure this to not have to write to `bucketedFeatureVariations` and iterate over it again, as we are adding some allocations and needing to loop over this new array.

Reply:
updated